File: Core Data/Core Data Sample Codes/iPhoneCoreDataRecipes/iPhoneCoreDataRecipes/Classes/RecipePhotoViewController.m
/*
 Copyright (C) 2017 Apple Inc. All Rights Reserved.
 See LICENSE.txt for this sample’s licensing information
 
 Abstract:
 View controller to manage a view to display a recipe's photo.
  The image view is created programmatically.
 */

#import "RecipePhotoViewController.h"
#import "Recipe.h"

@interface RecipePhotoViewController ()

@property(nonatomic, strong) UIImageView *imageView;

@end


#pragma mark -

@implementation RecipePhotoViewController

- (void)viewDidLoad {
    
    [super viewDidLoad];
    
    self.title = NSLocalizedString(@"Photo", @"");
    
    _imageView = [[UIImageView alloc] initWithFrame:[UIScreen mainScreen].bounds];
    self.imageView.contentMode = UIViewContentModeScaleAspectFit;
    self.imageView.backgroundColor = [UIColor blackColor];
    
    self.view = self.imageView;
}

- (void)viewWillAppear:(BOOL)animated {
    
    [super viewWillAppear:animated];
    self.imageView.image = [self.recipe.image valueForKey:@"image"];
}

@end

File: Core Data/Core Data Sample Codes/iPhoneCoreDataRecipes/Recipes_Swift_3_Updated/Recipes_Swift/Recipe View Controllers/RecipePhotoViewController.swift
/*
  RecipePhotoViewController.swift
  Recipes_Swift

  Created by Seyed Samad Gholamzadeh on 2/2/18.
  Copyright © 2018 Seyed Samad Gholamzadeh. All rights reserved.

Abstract:
View controller to manage a view to display a recipe's photo.
The image view is created programmatically.
*/


import UIKit

class RecipePhotoViewController: UIViewController {

	var recipe: Recipe!
	var imageView: UIImageView!
	
    override func viewDidLoad() {
        super.viewDidLoad()
		
		self.title = NSLocalizedString("Photo", comment: "")
		
		self.imageView = UIImageView(frame: UIScreen.main.bounds)
		self.imageView.contentMode = .scaleAspectFit
		self.imageView.backgroundColor = UIColor.black
		
		self.view = self.imageView
    }
	
	override func viewWillAppear(_ animated: Bool) {
		super.viewWillAppear(animated)
		
		self.imageView.image = self.recipe.image?.image as! UIImage
	}

}
